### ISSUE TITLE
error in migrating the event publisher folder in ISKeyManager

### ISSUE THREAD

**Issue by shavantha** [wso2/product-apim#3169]
The below error was thrown when migrating the ISKM 5.3.0 to 5.5.0

`Error occurred while migrating data in folder : /wso2is-km-5.5.0/repository/deployment/server/eventpublishers . `

```
 : SysLogPropertiesMigrator starting........................... 
[2018-04-04 17:04:09,361]  INFO {org.wso2.carbon.is.migration.service.v550.migrator.SysLogPropertiesMigrator} -   WSO2 Product Migration Service Task : Migration starting on SYSLOG_PROPERTIES file
[2018-04-04 17:04:09,427]  INFO {org.wso2.carbon.is.migration.service.v550.RegistryDataManager} -  Sys log property password migrated for tenant : carbon.super
[2018-04-04 17:04:09,433]  INFO {org.wso2.carbon.is.migration.VersionMigration} -   WSO2 Product Migration Service Task : Version : 5.5.0, Migration Step : SchemaMigrator starting........................... 
[2018-04-04 17:04:09,434]  INFO {org.wso2.carbon.is.migration.service.SchemaMigrator} -   WSO2 Product Migration Service Task : Executing Identity Migration Scripts.
[2018-04-04 17:04:09,573]  INFO {org.wso2.carbon.is.migration.service.SchemaMigrator} -   WSO2 Product Migration Service Task : Identity DB Migration script executed successfully.
[2018-04-04 17:04:09,625]  INFO {org.wso2.carbon.is.migration.VersionMigration} -   WSO2 Product Migration Service Task : Version : 5.5.0, Migration Step : PolicySubscriberDataMigrator starting........................... 
[2018-04-04 17:04:09,625]  INFO {org.wso2.carbon.is.migration.service.v550.migrator.PolicySubscriberDataMigrator} -   WSO2 Product Migration Service Task : Migration starting on Subscriber resources
[2018-04-04 17:04:11,033]  INFO {org.wso2.carbon.is.migration.service.v550.RegistryDataManager} -  Policy Subscribers migrated for tenant : carbon.super
[2018-04-04 17:04:11,039]  INFO {org.wso2.carbon.is.migration.VersionMigration} -   WSO2 Product Migration Service Task : Version : 5.5.0, Migration Step : KeyStorePasswordMigrator starting........................... 
[2018-04-04 17:04:11,039]  INFO {org.wso2.carbon.is.migration.service.v550.migrator.KeyStorePasswordMigrator} -   WSO2 Product Migration Service Task : Migration starting on Key Stores
[2018-04-04 17:04:12,070]  INFO {org.wso2.carbon.is.migration.service.v550.RegistryDataManager} -  Keystore passwords migrated for tenant : carbon.super
[2018-04-04 17:04:12,075]  INFO {org.wso2.carbon.is.migration.VersionMigration} -   WSO2 Product Migration Service Task : Version : 5.5.0, Migration Step : SecurityPolicyPasswordMigrator starting........................... 
[2018-04-04 17:04:12,076]  INFO {org.wso2.carbon.is.migration.service.v550.migrator.SecurityPolicyPasswordMigrator} -   WSO2 Product Migration Service Task : Migration starting on Security Policies
[2018-04-04 17:04:12,167]  INFO {org.wso2.carbon.is.migration.service.v550.RegistryDataManager} -  Policy Subscribers migrated for tenant : carbon.super
[2018-04-04 17:04:12,172]  INFO {org.wso2.carbon.is.migration.VersionMigration} -   WSO2 Product Migration Service Task : Version : 5.5.0, Migration Step : EventPublisherMigrator starting........................... 
[2018-04-04 17:04:12,173]  INFO {org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator} -   WSO2 Product Migration Service Task : Migration starting on event publisher files. 
[2018-04-04 17:04:12,181] ERROR {org.wso2.carbon.is.migration.MigrationClientImpl} -  Migration process was stopped.
org.wso2.carbon.identity.core.migrate.MigrationClientException: Error occurred while migrating data in folder : /home/shavantha/products/testproject/testfolder/release100/wso2is-km-5.5.0/repository/deployment/server/eventpublishers . 
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.migrateData(EventPublisherMigrator.java:91)
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.migrateEventPublishers(EventPublisherMigrator.java:60)
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.migrate(EventPublisherMigrator.java:53)
	at org.wso2.carbon.is.migration.VersionMigration.migrate(VersionMigration.java:41)
	at org.wso2.carbon.is.migration.MigrationClientImpl.execute(MigrationClientImpl.java:80)
	at org.wso2.carbon.identity.core.internal.IdentityCoreServiceComponent.activate(IdentityCoreServiceComponent.java:147)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
	at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:107)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:861)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:819)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:771)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:214)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:433)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:451)
	at org.wso2.carbon.core.init.CarbonServerManager.initializeCarbon(CarbonServerManager.java:515)
	at org.wso2.carbon.core.init.CarbonServerManager.start(CarbonServerManager.java:220)
	at org.wso2.carbon.core.internal.CarbonCoreServiceComponent.activate(CarbonCoreServiceComponent.java:94)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:260)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
	at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:107)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:861)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:819)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:771)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:214)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:433)
	at org.eclipse.equinox.http.servlet.internal.Activator.registerHttpService(Activator.java:81)
	at org.eclipse.equinox.http.servlet.internal.Activator.addProxyServlet(Activator.java:60)
	at org.eclipse.equinox.http.servlet.internal.ProxyServlet.init(ProxyServlet.java:40)
	at org.wso2.carbon.tomcat.ext.servlet.DelegationServlet.init(DelegationServlet.java:38)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1230)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1174)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1066)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:5370)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5668)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1700)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1690)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.wso2.carbon.identity.core.migrate.MigrationClientException: Error has occurred while retriving the payload from file : file:///home/shavantha/products/testproject/testfolder/release100/wso2is-km-5.5.0/repository/deployment/server/eventpublishers/IsAnalytics-Publisher-wso2event-UserData.xml
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.getEncryptedPayload(EventPublisherMigrator.java:108)
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.migrateData(EventPublisherMigrator.java:75)
	... 65 more
Caused by: javax.xml.xpath.XPathExpressionException: net.sf.saxon.trans.XPathException: Cannot compare xs:boolean to xs:string
	at net.sf.saxon.xpath.XPathEvaluator.compile(XPathEvaluator.java:378)
	at org.wso2.carbon.is.migration.service.v550.migrator.EventPublisherMigrator.getEncryptedPayload(EventPublisherMigrator.java:104)
	... 66 more
Caused by: net.sf.saxon.trans.XPathException: Cannot compare xs:boolean to xs:string
	at net.sf.saxon.expr.GeneralComparison.typeCheck(GeneralComparison.java:245)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.FilterExpression.typeCheck(FilterExpression.java:220)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.FilterExpression.typeCheck(FilterExpression.java:220)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.SlashExpression.typeCheck(SlashExpression.java:151)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.UnaryExpression.typeCheck(UnaryExpression.java:54)
	at net.sf.saxon.expr.HomogeneityChecker.typeCheck(HomogeneityChecker.java:34)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.SlashExpression.typeCheck(SlashExpression.java:140)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.expr.UnaryExpression.typeCheck(UnaryExpression.java:54)
	at net.sf.saxon.expr.HomogeneityChecker.typeCheck(HomogeneityChecker.java:34)
	at net.sf.saxon.expr.parser.ExpressionVisitor.typeCheck(ExpressionVisitor.java:206)
	at net.sf.saxon.xpath.XPathEvaluator.createExpressionInternal(XPathEvaluator.java:239)
	at net.sf.saxon.xpath.XPathEvaluator.compile(XPathEvaluator.java:376)
	... 67 more
[2018-04-04 17:04:12,187]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:12,188]  INFO {org.wso2.carbon.is.migration.MigrationClientImpl} -  ............................................................................................
[2018-04-04 17:04:13,810]  INFO {org.wso2.carbon.user.core.internal.UserStoreMgtDSComponent} -  Claim manager set for class org.wso2.carbon.user.core.common.DefaultRealm
[2018-04-04 17:04:13,811]  INFO {org.wso2.carbon.user.core.internal.UserStoreMgtDSComponent} -  Claim manager set for class org.wso2.carbon.user.core.jdbc.JDBCUserStoreManager
[2018-04-04 17:04:14,059]  INFO {org.wso2.carbon.identity.oauth2.internal.OAuth2ServiceComponent} -  PKCE Support enabled.
[2018-04-04 17:04:14,652]  INFO {org.wso2.carbon.identity.entitlement.internal.SchemaBuilder} -  XACML policy schema loaded successfully.
[2018-04-04 17:04:18,890] ERROR {org.wso2.carbon.identity.entitlement.EntitlementUtil} -  Error while adding sample XACML policies
java.lang.IllegalArgumentException: Error while parsing start up policy
	at org.wso2.carbon.identity.entitlement.EntitlementUtil.getPolicy(EntitlementUtil.java:503)
	at org.wso2.carbon.identity.entitlement.EntitlementUtil.addFilesystemPolicy(EntitlementUtil.java:436)
	at org.wso2.carbon.identity.entitlement.EntitlementUtil.addSamplePolicies(EntitlementUtil.java:391)
	at org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent.activate(EntitlementServiceComponent.java:279)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
	at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:107)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:861)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:819)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:771)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:214)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:433)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:451)
	at org.wso2.carbon.identity.core.internal.IdentityCoreServiceComponent.activate(IdentityCoreServiceComponent.java:171)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
	at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:107)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:861)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:819)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:771)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:214)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:433)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:451)
	at org.wso2.carbon.core.init.CarbonServerManager.initializeCarbon(CarbonServerManager.java:515)
	at org.wso2.carbon.core.init.CarbonServerManager.start(CarbonServerManager.java:220)
	at org.wso2.carbon.core.internal.CarbonCoreServiceComponent.activate(CarbonCoreServiceComponent.java:94)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:260)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.getEligible(Resolver.java:343)
	at org.eclipse.equinox.internal.ds.SCRManager.serviceChanged(SCRManager.java:222)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:107)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:861)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:819)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:771)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:130)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:214)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.registerService(BundleContextImpl.java:433)
	at org.eclipse.equinox.http.servlet.internal.Activator.registerHttpService(Activator.java:81)
	at org.eclipse.equinox.http.servlet.internal.Activator.addProxyServlet(Activator.java:60)
	at org.eclipse.equinox.http.servlet.internal.ProxyServlet.init(ProxyServlet.java:40)
	at org.wso2.carbon.tomcat.ext.servlet.DelegationServlet.init(DelegationServlet.java:38)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1230)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1174)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1066)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:5370)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5668)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1700)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1690)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.wso2.balana.ParsingException: Unknown FunctionId
	at org.wso2.balana.cond.ExpressionHandler.getFunction(ExpressionHandler.java:102)
	at org.wso2.balana.cond.Apply.getInstance(Apply.java:216)
	at org.wso2.balana.cond.Apply.getInstance(Apply.java:188)
	at org.wso2.balana.cond.ExpressionHandler.parseExpression(ExpressionHandler.java:53)
	at org.wso2.balana.cond.Condition.getInstance(Condition.java:177)
	at org.wso2.balana.Rule.getInstance(Rule.java:237)
	at org.wso2.balana.Policy.<init>(Policy.java:303)
	at org.wso2.balana.Policy.getInstance(Policy.java:382)
	at org.wso2.carbon.identity.entitlement.EntitlementUtil.getPolicy(EntitlementUtil.java:495)
	... 85 more
Caused by: org.wso2.balana.UnknownIdentifierException: functions of type urn:oasis:names:tc:xacml:1.0:function:eval-permission-tree are not supported by this factory
	at org.wso2.balana.cond.BaseFunctionFactory.createFunction(BaseFunctionFactory.java:240)
	at org.wso2.balana.cond.ExpressionHandler.getFunction(ExpressionHandler.java:100)
	... 93 more
[2018-04-04 17:04:21,523]  INFO {org.wso2.carbon.identity.sso.saml.admin.FileBasedConfigManager} -  A SSO Service Provider is registered for : wso2.my.dashboard


```

**Comment by shavantha** [comment]
Doc Impact :N/A

**Comment by tgtshanika** [comment]
Fixed in APIM 2.6.0
